pkg/utils/kubeconfig/kubeconfig.go

```diff
@@ -180,6 +206,37 @@ func AppendAuthenticator(config *clientcmdapi.Config, clusterMeta *api.ClusterMe
 	}
 }
 
+// AWSAuthenticatorVersionFormat is the format in which aws-iam-authenticator displays version information:
+// {"Version":"0.5.5","Commit":"85e50980d9d916ae95882176c18f14ae145f916f"}
+type AWSAuthenticatorVersionFormat struct {
+	Version string `json:"Version"`
+}
+
+func authenticatorIsAboveVersion(version string) (bool, error) {
+	authenticatorVersion, err := CheckAuthenticatorVersion()
+	if err != nil {
+		return false, fmt.Errorf("failed to retrieve authenticator version: %w", err)
+	}
+	compareVersions, err := utils.CompareVersions(authenticatorVersion, version)
+	if err != nil {
+		return false, fmt.Errorf("failed to parse versions: %w", err)
+	}
+	return compareVersions >= 0, nil
+}
+
+func getAWSIAMAuthenticatorVersion() (string, error) {
+	cmd := exec.Command(AWSIAMAuthenticator, "version")
+	output, err := cmd.Output()
+	if err != nil {
+		return "", fmt.Errorf("failed to run aws-iam-authenticator version command: %w", err)
+	}
+	var parsedVersion AWSAuthenticatorVersionFormat
+	if err := json.Unmarshal(output, &parsedVersion); err != nil {
+		return "", fmt.Errorf("failed to parse version information: %w", err)
+	}
+	return parsedVersion.Version, nil
+}
```

Comment:
so this whole func isn't tested right? Would it not be possible to test this with unit tests using some fake binaries if mocking `exec` is too difficult

Reply:
I don't want a unit test to mess with a user's environment such as possibly modify the PATH to include a new binary.

Reply:
mhmm there must be some way of testing this code, it seems quite important

Reply:
I mean the error is essentially ignored, it's just logged. I didn't want to block execution on this. I could mock out the entire exec.Command things and pass it in and wire it up before execution. But honestly, that feels like a massive overkill.

Reply:
Oh! I actually remembered something with an environment variable that I can set to test a shell command. I shall do some thing. :D But it might not be with Ginkgo. 🤔 I see what I can do. :)

Reply:
Yeah, that works by overriding the runtime and using a`-run=` to run a specific test. So that's not gonna work with ginkgo.

Reply:
Ah!!!!!! I have such an awesome solution. :D

Reply:
@aclevername BOOM! How do you like THEM APPELS? :D

Reply:
me gusta